admin/metabox/class-metabox.php

```diff
@@ -335,6 +331,22 @@ private function get_seo_meta_section() {
 			$label = '<span class="wpseo-score-icon-container" id="wpseo-seo-score-icon"></span>' . $label;
 		}
 
+		$html_after = '';
+
+		if ( WPSEO_Capability_Utils::current_user_can( 'wpseo_edit_advanced_metadata' ) || WPSEO_Options::get( 'disableadvanced_meta' ) === false ) {
```

Comment:
Maybe you can consider putting this check in its own methods because you are using the same logic at another location.